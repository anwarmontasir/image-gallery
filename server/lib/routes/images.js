const Router = require('express').Router;
const router = Router();
const Image = require('../models/Image');

router
    .get('/', (req, res, next) => {
        Image.find()
            .lean()
            .select('title description url')
            .then(images => res.send(images))
            .catch(next);
    })

    .post('/', (req, res, next) => {
        new Image(req.body)
            .save()
            .then(image => res.send(image))
            .catch(next);
    });

module.exports = router;