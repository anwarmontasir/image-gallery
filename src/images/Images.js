// import React from 'react';

// export default function Images({ match, location, props }) {
//     <header>
//         <h1>Cute Bunnies: The App</h1>
//     </header>
//         <nav>
//             {views.map(v => (
//                 <button key={v} onClick={() => this.setState({ view: v })}>
//                     {v}
//                 </button>
//             ))}
//         </nav>
//         <section>
//             <ViewWrapper bunnies={bunnies} bunnyNum={props.bunnyNum} onRemove={props.removeBunny} onUpdate={props.updateBunny} />
//         </section>
//         <section>
//             <AddBunny onAdd={props.addBunny} />
//         </section>

// }