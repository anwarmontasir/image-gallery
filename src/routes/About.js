import React from 'react';

export default function About({ match, location, history }) {
    return (
        <div>
            <p>This bunny app was created with care by Anwar Montasir and Stephanie Fitzgerald. We like bunnies. We like cats better.</p>
        </div>
    );
}