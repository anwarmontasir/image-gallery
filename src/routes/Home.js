import React from 'react';

export default function Home({ match, location, history }) {
    return (
        <div>
            <h1>Bunny App Home Placeholder</h1>
        </div>
    );
}