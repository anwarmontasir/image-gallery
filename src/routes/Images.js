import React, { Component } from 'react';
import AddBunny from '../components/images/AddBunny';
import { List } from '../components/images/List';
import { Thumbs } from '../components/images/Thumbs';
import { Gallery } from '../components/images/Gallery';
import { Link } from 'react-router-dom';
import qs from 'qs';

function listView({ bunnies, onRemove }) {
    return <div><h2>List View</h2> <List listArray={bunnies} onRemove={onRemove} /> </div>;
}
function thumbView({ bunnies, onRemove }) {
    return <div><h2>Thumbnail View</h2> <Thumbs thumbArray={bunnies} onRemove={onRemove} /> </div>;
}
function galleryView({ bunnies, onRemove, onUpdate, bunnyNum }) {
    return <div><h2>Gallery View</h2> <Gallery galleryArray={bunnies} bunnyNum={bunnyNum} onRemove={onRemove} onUpdate={onUpdate} /> </div>;
}

const viewDict = {
    list: listView,
    thumbs: thumbView,
    gallery: galleryView
}

const viewArray = Object.keys(viewDict);

export class Images extends Component {

    constructor(props) {
        super(props);
        this.state = {
            bunnies: []
        }
    }

    componentDidMount() {
        fetch('/api/images')
            .then(res => res.json())
            .then(bunnies => this.setState({ bunnies }))
            .catch(error => console.log(error));
    }

    render() {
        const { bunnies } = this.state;
        const { match, location } = this.props;
        const view = qs.parse(location.search.slice(1)).view;
        const ViewWrapper = viewDict[view];

        return (
            <div>
                <nav id="subNav">
                    <ul>
                        {
                            viewArray.map(v =>
                                <li key={v}><Link to={`${match.url}?view=${v}`}>
                                    {v}</Link></li>
                            )
                        }
                    </ul>
                </nav>
                <section>
                    <ViewWrapper bunnies={bunnies} />
                </section>
                <section>
                    <AddBunny />
                </section>
            </div>
        );
    }
}
