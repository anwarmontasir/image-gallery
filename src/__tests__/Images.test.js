import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import toJSON from 'enzyme-to-json';

import { Images } from '../routes/Images';

describe('Images tests', () => {
    it('shows view that matches user selection', () => {
        const wrapper = shallow(<Images />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

// TODO: do we need a handleChange function so we can simulate clicking on the other views, in order to test this?
})