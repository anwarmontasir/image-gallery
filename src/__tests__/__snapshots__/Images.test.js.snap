// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images tests shows view that matches user selection 1`] = `
<div>
  <nav>
    <button
      onClick={[Function]}
    >
      list
    </button>
    <button
      onClick={[Function]}
    >
      thumbs
    </button>
    <button
      onClick={[Function]}
    >
      gallery
    </button>
  </nav>
  <section>
    <listView
      bunnies={
        Array [
          Object {
            "description": "Who's a cute bunny??",
            "title": "Cute Bunny 1",
            "url": "http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg",
          },
          Object {
            "description": "Diabolically cute!!",
            "title": "Cute Bunny 2",
            "url": "http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg",
          },
          Object {
            "description": "There are no bounds to the cuteness!",
            "title": "Cute Bunny 3",
            "url": "http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg",
          },
          Object {
            "description": "Hey, that's not a bunny!",
            "title": "Cute Bunny 4",
            "url": "https://i.pinimg.com/originals/52/79/ec/5279ec99c74876b17224e0e033224222.jpg",
          },
        ]
      }
      bunnyNum={0}
      onRemove={[Function]}
      onUpdate={[Function]}
    />
  </section>
  <section>
    <AddBunny
      onAdd={[Function]}
    />
  </section>
</div>
`;
